lowcodelanguage/src/main/java/net/minecraftforge/fml/lowcodemod/LowCodeModLanguageProvider.java

```diff
@@ -0,0 +1,82 @@
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static net.minecraftforge.fml.Logging.LOADING;
+
+public class LowCodeModLanguageProvider implements IModLanguageProvider
+{
+    private record LowCodeModTarget(String modId) implements IModLanguageProvider.IModLanguageLoader {
+            private static final Logger LOGGER = LogManager.getLogger();
+
+        @SuppressWarnings("unchecked")
+            @Override
+            public <T> T loadMod(final IModInfo info, final ModFileScanData modFileScanResults, ModuleLayer gameLayer) {
+                // This language class is loaded in the system level classloader - before the game even starts
+                // So we must treat container construction as an arms length operation, and load the container
+                // in the classloader of the game - the context classloader is appropriate here.
+                try {
+                    final Class<?> fmlContainer = Class.forName("net.minecraftforge.fml.lowcodemod.LowCodeModContainer", true, Thread.currentThread().getContextClassLoader());
+                    LOGGER.debug(LOADING, "Loading LowCodeModContainer from classloader {} - got {}", Thread.currentThread().getContextClassLoader(), fmlContainer.getClassLoader());
+                    final Constructor<?> constructor = fmlContainer.getConstructor(IModInfo.class, ModFileScanData.class, ModuleLayer.class);
+                    return (T) constructor.newInstance(info, modFileScanResults, gameLayer);
+                } catch (InvocationTargetException e) {
+                    LOGGER.fatal(LOADING, "Failed to build mod", e);
+                    if (e.getTargetException() instanceof ModLoadingException mle) {
+                        throw mle;
+                    } else {
+                        throw new ModLoadingException(info, ModLoadingStage.CONSTRUCT, "fml.modloading.failedtoloadmodclass", e);
+                    }
+                } catch (NoSuchMethodException | ClassNotFoundException | InstantiationException |
+                         IllegalAccessException e) {
+                    LOGGER.fatal(LOADING, "Unable to load FMLModContainer, wut?", e);
+                    final Class<RuntimeException> mle = (Class<RuntimeException>) LamdbaExceptionUtils.uncheck(() -> Class.forName("net.minecraftforge.fml.ModLoadingException", true, Thread.currentThread().getContextClassLoader()));
+                    final Class<ModLoadingStage> mls = (Class<ModLoadingStage>) LamdbaExceptionUtils.uncheck(() -> Class.forName("net.minecraftforge.fml.ModLoadingStage", true, Thread.currentThread().getContextClassLoader()));
+                    throw LamdbaExceptionUtils.uncheck(() -> LamdbaExceptionUtils.uncheck(() -> mle.getConstructor(IModInfo.class, mls, String.class, Throwable.class)).newInstance(info, Enum.valueOf(mls, "CONSTRUCT"), "fml.modloading.failedtoloadmodclass", e));
+                }
+            }
+        }
```

Comment:
Is there any way to deduplicate some of this code compared to `FMLJavaModLanguageProvider`? Additionally, this file also has a mix of formatting.

Reply:
No, deduping is not possible, because they're literally in separate entire projects (effectively). To introduce a common library would break LITERALLY EVERYTHING. Feel free to try!